### PR TITLE
[ci] fixed tests for building packages

### PR DESCRIPTION
**Pull request by M0ses**
* splitted test to make re-run easier
* added reconfiguration of number of workers to avoid full disk or
  running out of memory
* defined function obs_login to make it reusable in multiple scripts
* added DOD tests to avoid unresolvables
* disabled x86_64 in Tumbleweed to reduce testing time
* changed order of checking build results and checking build logs
  to reduce testing time if running into unresolvable
* increased timeout while waiting for build (in case of testing over
  umts)
* added code to wait for "State needs recalculation" to give bs_schedule
  time to recalculate states without running into unresolvable
* added deletion of home:Admin as a test at the end to make test cases
  re-runable

### CODE REVIEW

**Comment by M0ses** [comment]
@shyukri : Please review and comment

File: dist/t/spec/features/0000_set_workers_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require "spec_helper"
+#for getting spec file
+require 'tmpdir'
+require "net/https"
+require "uri"
+
```
Comment (shyukri) [inline dist/t/spec/features/0000_set_workers_spec.rb]: I think we need those 3 `(tmpdir, net/https, uri)` only on **Preparation for building package obs-build** spec

File: dist/t/spec/features/0060_build_result_spec.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    visit("/project/show/home:Admin")
+    click_link('Overview')
+    expect(page).to have_content("Build Results")
+  end
+
+  it "should be able to check Build Results and see succeeded package built" do
+    sleep(10)
+    while page.has_content?("State needs recalculation") do
+      sleep(3)
+      find("img[alt='Reload']").click
```
Comment (shyukri) [inline dist/t/spec/features/0060_build_result_spec.rb]: I didn't had good experience with this reload, especially on selenium.
Maybe direct call to JS reload would work better: `page.evaluate_script("window.location.reload()")`

**Comment by shyukri** [comment]
:+1:

**Pull request merged**